v2/internal/platform/systray/windows.go

```diff
@@ -0,0 +1,385 @@
+/*
+ * Based on code originally from https://github.com/tadvi/systray. Copyright (C) 2019 The Systray Authors. All Rights Reserved.
+ */
+
+package systray
+
+import (
+	"errors"
+	"github.com/samber/lo"
+	"github.com/wailsapp/wails/v2/internal/platform/win32"
+	"github.com/wailsapp/wails/v2/pkg/menu"
+	"github.com/wailsapp/wails/v2/pkg/options"
+	"syscall"
+	"unsafe"
+)
+
+var (
+	user32 = syscall.MustLoadDLL("user32.dll")
+
+	DefWindowProc   = user32.MustFindProc("DefWindowProcW")
+	RegisterClassEx = user32.MustFindProc("RegisterClassExW")
+	CreateWindowEx  = user32.MustFindProc("CreateWindowExW")
+
+	windowClasses = map[string]win32.HINSTANCE{}
+)
+
+type Systray struct {
+	id     uint32
+	mhwnd  win32.HWND // main window handle
+	hwnd   win32.HWND
+	hinst  win32.HINSTANCE
+	lclick func()
+	rclick func()
+
+	appIcon       win32.HICON
+	lightModeIcon win32.HICON
+	darkModeIcon  win32.HICON
+	currentIcon   win32.HICON
+
+	menu *PopupMenu
+
+	quit chan struct{}
+	icon *options.SystemTrayIcon
+}
+
+func (p *Systray) Close() {
+	err := p.Stop()
+	if err != nil {
+		println(err.Error())
+	}
+}
+
+func (p *Systray) Update() error {
+	// Delete old menu
+	if p.menu != nil {
+		p.menu.Destroy()
```

Comment:
Not sure if I’ve understood this correctly, but is the menu destroyed on each `Update()`?

I’m not familiar with Windows APIs, but it seems a tad wasteful to completely remove the old menu and recreate it on update. Is it possible to update the menu item objects and have them redraw when needed?

Reply:
I've debated this but so far I'm unconvinced that it's an optimisation worth the effort. How often do people do this? How much time or memory do you really save? A lot of APIs make the developer remember where menu items are based on array indexes and use that as a mechanism for updating the underlying item only. I'm unconvinced of that approach too.

Reply:
Thanks for the explanation. I agree that the current approach makes it simpler to update the menus.